rebasehelper/plugins/spec_hooks/replace_old_version.py

```diff
@@ -116,34 +115,35 @@ def run(cls, spec_file: SpecFile, rebase_spec_file: SpecFile, **kwargs: Any):
-            if sec_name.startswith('%changelog'):
-                continue
-            for index, line in enumerate(section):
-                tag_ignored = any(MacroHelper.expand(line, line).startswith(tag) for tag in cls.IGNORED_TAGS)
-                if index in examined_lines[sec_index] or tag_ignored:
+        with rebase_spec_file.spec.sections() as sections:
+            for section in sections:
+                if section.normalized_name.startswith('changelog'):
                     continue
-                start, end = spec_file.spec_content.get_comment_span(line, sec_name)
-                updated_line = subversion_patterns[0][0].sub(subversion_patterns[0][1], line[:start])
-                section[index] = updated_line + line[start:end]
-
-        rebase_spec_file.save()
+                elif section.normalized_name.startswith('package'):
+                    with rebase_spec_file.spec.tags(section) as tags:
+                        for tag in tags:
+                            for index, line in enumerate(tag.comments._preceding_lines): # pylint: disable=protected-access
+                                start, end = spec_file.get_comment_span(line, section.normalized_name)
+                                updated_line = subversion_patterns[0][0].sub(subversion_patterns[0][1], line[:start])
+                                tag.comments._preceding_lines[index] = updated_line + line[start:end] # pylint: disable=protected-access
+                            if not tag.value or tag.normalized_name in cls.IGNORED_TAGS:
+                                continue
+                            if (
+                                tag.normalized_name.startswith("Source")
+                                or tag.normalized_name.startswith("Patch")
+                            ) and not TagSource(tag).remote:
+                                # skip local sources
+                                continue
+                            # replace the whole version first
+                            updated_value = subversion_patterns[0][0].sub(subversion_patterns[0][1], tag.value)
+                            # replace subversions only for remote sources/patches
+                            if tag.normalized_name.startswith('Source') or tag.normalized_name.startswith('Patch'):
+                                for sub_pattern, repl in subversion_patterns[1:]:
+                                    updated_value = sub_pattern.sub(repl, updated_value)
+                            tag.value = updated_value
+                    continue
+                for index, line in enumerate(section):
+                    start, end = spec_file.get_comment_span(line, section.normalized_name)
+                    updated_line = subversion_patterns[0][0].sub(subversion_patterns[0][1], line[:start])
+                    section[index] = updated_line + line[start:end]
+            rebase_spec_file.save()
```

Comment:
Is save still necessary? Isn't this handled by autosave in the context manager?

Reply:
I kept it there because it was there originally, but it probably was unnecessary even then. I'll remove it.

Reply:
Actually, autosaving is not enabled and I'd rather keep the explicit save here.